### PR TITLE
Smd 427/feature/exception type logs

### PR DESCRIPTION
**Pull request by bobbi-hbrown**
https://dluhcdigital.atlassian.net/jira/software/c/projects/SMD/boards/140?quickFilter=815&selectedIssue=SMD-427

This draft PR just displays the data-store changes from the same ticket - the formatting can be changed depending on logging requirements. Related PR is https://github.com/communitiesuk/funding-service-design-post-award-data-store/pull/367

### CODE REVIEW

File: app/main/routes.py [hunk 1]
```diff
@@ -55,7 +63,7 @@ def upload():
                 fund=Config.FUND_NAME,
             )
 
-        success, pre_errors, validation_errors, metadata = post_ingest(
+        success, pre_errors, validation_errors, metadata, error_types = post_ingest(
```
Comment (Joel-Pearce) [inline app/main/routes.py:66]: Maybe we can rename ```pre_errors``` to ```pre_transformation_errors``` or something similar to match how the variable is named in ```data_requests.py```

**Comment by cyrusdobbs** [comment]
As mentioned in https://github.com/communitiesuk/funding-service-design-post-award-data-store/pull/367, I think we can avoid all of these changes if we insert the extra information in the existing failure JSON objects. Would be good to update the tests to reflect the change in the ingest interface though and do some testing to ensure that the templates still work as expected with the additional field in the failure JSON.

**Comment by cyrusdobbs** [comment]
Just a reminder that the openapi documentation should also be updated in line with this change:

https://github.com/communitiesuk/funding-service-design-post-award-data-store/blob/96ce32bc6bb2e5c8d45fc1506eadffa059752350/openapi/components.yml#L165-L182

**Review by cyrusdobbs — COMMENTED** [review]
Probably best to clean up the git history on this PR now that the change is minimal

File: tests/test_routes.py [hunk 2]
```diff
@@ -99,6 +101,15 @@ def test_upload_xlsx_validation_errors(requests_mock, example_pre_ingest_data_fi
...
 
+    # assert errors display correctly
+    assert '<th class="govuk-table__header" scope="col">Description</th>' in str(page_html)
+    assert '<th class="govuk-table__header" scope="col">Cell</th>' in str(page_html)
+    assert (
+        '<td class="govuk-table__cell">You are missing project locations. Please enter a project location.</td>'
+        in str(page_html)
+    )
+    assert '<td class="govuk-table__cell">Section2</td>' in str(page_html)
+
```
Comment (cyrusdobbs) [inline tests/test_routes.py:112]: I don't think we need to make these extra assertions as they're just testing the static template. The existing assertions already check that the template is rendered and that the errors are displayed in some way.

The addition of the `error_type`s to the mock ingest payload is useful though. Now we know that their inclusion won't cause the template to fail being rendered.

**Review by cyrusdobbs — APPROVED** [review]
LGTM

**Pull request merged**